web/views/api.py

```diff
@@ -147,6 +149,38 @@ def datasets(request):
             {'more': str(e)}
         )
 
+
+def is_valid_field_in_model(klass_name, field_name):
+    getattr(klass_name, field_name, False)
+
+def get_filtered_entities(request, model_name):
+    filters = {}
+    for filter_key in request.GET:
+        if not is_valid_field_in_model(model_name, filter_key): continue
+        filters[filter_key] = request.GET.get(filter_key)
+
+    return getattr(sys.modules['core.models'], model_name).objects.filter(**filters)
```

Comment:
this is super generic :) 
I actually thought more about passing a concrete `Contract` class to the function (see line 168), but this is even better!

Reply:
I am not really sure we should be that generic :D just a bad feeling it makes things too loose.